staging/src/k8s.io/apimachinery/pkg/api/testing/roundtrip/roundtrip.go

```diff
@@ -207,14 +213,18 @@ func roundTripToAllExternalVersions(t *testing.T, scheme *runtime.Scheme, codecF
 		if externalGVK.Version == runtime.APIVersionInternal {
 			continue
 		}
-		if externalGVK.GroupKind() != internalGVK.GroupKind() {
+		if externalGVK.Kind != internalGVK.Kind {
+			continue
+		}
+		if nonRoundTrippableTypes[externalGVK] || unrelated[UnrelatedKinds{internalGVK.GroupKind(), externalGVK.Group}] {
+			t.Logf("\tskipping  %v", externalGVK)
 			continue
 		}
-		if nonRoundTrippableTypes[externalGVK] {
-			t.Logf("\tskipping  %v %v", externalGVK, externalGoType)
+		if !scheme.Converter().HasConversionFunc(objectType, externalGoType) {
+			t.Logf("\tno conversion to %v", externalGVK)
```

Comment:
Because we have plenty of errors right now: https://gist.github.com/sttts/99acd67797f430bcddbb6251b4af36b2